### ISSUE TITLE
Reconsider media object types

### ISSUE THREAD

**Issue by RemcoSimonides** [blockframes/blockframes#4241]
The amount of different media types is increasing which makes the code increasingly hard to maintain. Would be nice if there is a way to homogenize the types more.

Current types:
Still_photo: string[] = `[]`
otherVideos: HostedVideo[] = `{ description, jwPlayerId, ref, title, type }[]`
screener: HostedVideo = `{ description, jwPlayerId, ref, title, type }`
logo: string = `''`
document: HostedMediaWithMetadata[] = `{ title, ref }[]`
Notes & Statements: MovieNote = `{ firstName, lastName, ref, role }`

p.s. there are more media field with types mentioned above

**Comment by RemcoSimonides** [comment]
It would be nice to have this structure:
`[tmp (optional)] / [privacy] / [collection] / [docId] / [sub collection (optional)] / [docId] / [field] / [fileName]`
Where `field` is never split by a `/`

Also, for an array of files, it would be nice to have an identifier to find each file. Currently in the code we search for matching reference to find the element. But users could for example use the same video file name for different kind of videos.

**Comment by RemcoSimonides** [comment]
Hee @phpgeekfr, do you mind if we remove `description` and `type` from the HostedVideo object? I don't think it is used at the moment.

**Comment by RemcoSimonides** [comment]
Also, the code would be easier if we have a `{ ref: string }` object for logo and and still_photo array. Currently in the code we check a lot whether `typeof media === 'string' ? media : media.ref`. If everything has ref, it would make the code more readable I think

**Comment by phpgeekfr** [comment]
> Hee @phpgeekfr, do you mind if we remove `description` and `type` from the HostedVideo object? I don't think it is used at the moment.

Yes if this is not used, you can remove it

**Comment by phpgeekfr** [comment]
> Also, the code would be easier if we have a `{ ref: string }` object for logo and and still_photo array. Currently in the code we check a lot whether `typeof media === 'string' ? media : media.ref`. If everything has ref, it would make the code more readable I think

If this is not to much time ( but I think it will ..) yeah But lets discuss it with @GrandSchtroumpf  before starting this

**Comment by RemcoSimonides** [comment]
GrandSchtroumpf said to tag the occurances of this issue and see if it's worth to change it. It might not be a real issue yet. If it's more than 4 then we could use this ref object.

**Comment by RemcoSimonides** [comment]
I found 5 occurances

**Comment by RemcoSimonides** [comment]
Closed the issue because removing code like `typeof ref === string ? ref : ref.ref` will add more complexity somewhere else. This issue can always be reopened once more occurances are added.

See PR https://github.com/blockframes/blockframes/pull/4642 for current occurances.
